pygeos/test/test_creation_indices.py

```diff
@@ -61,16 +61,26 @@ def test_points_invalid():
 
 
 def test_points():
-    actual = pygeos.points([[2, 3], [2, 3]], indices=[0, 2])
-    assert_geometries_equal(actual, [point, None, point])
+    actual = pygeos.points(
+        np.array([[2, 3], [2, 3]], dtype=float),
+        indices=np.array([0, 1], dtype=np.intp),
+    )
+    assert_geometries_equal(actual, [point, point])
+
+
+def test_points_no_index_raises():
+    with pytest.raises(ValueError):
+        pygeos.points(
+            np.array([[2, 3], [2, 3]], dtype=float),
+            indices=np.array([0, 2], dtype=np.intp),
```

Comment:
```suggestion
            indices=np.array([0], dtype=np.intp),
```

(to keep it focused on only the missing `1` index; having an index out of bounds is separate problem)

Also, shouldn't this have failed because the arrays are of different sizes (i.e., this fails for `geometrycollections` and `polygons`)?

Reply:
No the idea of this test is that index `1` is missing here. If I would make the indices array `[0]` I would get a different error (that indices and coordinates are not of equal length)